katana.py

```diff
@@ -0,0 +1,21 @@
+import sys
+import collections
+
+from core.notify import Notify
+
+def main():
+    arg_names = ['title', 'message', 'app_name', 'app_icon']
+
+    args = collections.namedtuple('data', arg_names)(
+        *(dict(zip(arg_names, sys.argv[1:])).get(arg, "") for arg in arg_names))
```

Comment:
Seria melhor fazer umas variáveis para ficar menos complexo, ou é comum isso dessa maneira ?

Reply:
piá, essa ai é as bruxaria que tem no python. huahuhaua.
basicamente, eu tenho: 
 - `arg_names` com os tipos de argumentos que eu espero receber.
 - `namedtuple` que é uma tupla que atua como um objeto
 - `sys.argv` que é uma matriz cujos argumentos foram passados ​​para o katana.py

Ai basicamente eu faço um laço, passando os valores do array, verificando se os parâmetros foram preenchidos, e caso não tenha sido, eu atribuo `""` para eles.

Obs.: 
 - `*` - Significa que eu quero todos os itens
 - `dict` - Faz um "cast" para dicionario
 - `zip` - Junta as 2 listas, (valor e resultado), nesse caso em um dicionario de chave/valor
 - `get` - recebe da matriz, e retorna o valor padrão, nesse caso `""`, quando não encontrar um valor.


Confesso que ficaria mais claro se eu separar isso em 3 linhas. mas eu tava na vibe de fazer em uma só =/



Referencia: https://stackoverflow.com/questions/5423381/checking-if-sys-argvx-is-defined